nutils/matrix/__init__.py

```diff
@@ -89,4 +89,6 @@ def Scipy():
 def MKL(threading:str=None):
   return _helper('MKL', threading)
 
+backend(os.environ.get('NUTILS_MATRIX') or 'auto')
```

Comment:
Did you test this? Backend returns an unentered context so I don't think this is currently effective. Rather than entering it I would propose to set the _assemble default based on the envvar: `_assemble = util.settable(_import_backend(os.environ.get('NUTILS_MATRIX', 'auto')).assemble)`. The only downside is that we cannot pass the threading argument to mkl, but that might be solved using the MKL_THREADING_LAYER envvar: https://software.intel.com/content/www/us/en/develop/documentation/mkl-macos-developer-guide/top/linking-your-application-with-the-intel-math-kernel-library/linking-in-detail/dynamically-selecting-the-interface-and-threading-layer.html (not tested).